tests/L0/run_transformer/test_parallel_state.py

```diff
@@ -132,6 +132,50 @@ def test_initialize_model_parallel_with_virtual_and_split(self) -> None:
 
         parallel_state.destroy_model_parallel()
 
+    def test_initialize_model_parallel_decoder_only(self) -> None:
+        """Initialize model parallelism for decoder-only Transformers like GPT-3"""
+
+        self.assertFalse(parallel_state.model_parallel_is_initialized())
+
+        for tensor_model_parallel_world_size in range(1, self.world_size + 1):
+            with self.subTest(
+                tensor_model_parallel_world_size=tensor_model_parallel_world_size
+            ):
+                if self.world_size % tensor_model_parallel_world_size:
+                    continue
+
+                pipeline_model_parallel_world_size = (
+                    self.world_size // tensor_model_parallel_world_size
+                )
+
+                parallel_state.initialize_model_parallel(
+                    tensor_model_parallel_size_=tensor_model_parallel_world_size,
+                    pipeline_model_parallel_size_=pipeline_model_parallel_world_size,
+                    pipeline_model_parallel_split_rank_=0,
```

Comment:
out of curiosity, when do you set split rank to 0? I'm not sure when GPT like models (= ModelType.encoder_or_decoder) would require this value to 0.

Reply:
I see that the value is 0 when I print inside `initialize_model_parallel`. It seems that the default value is set in NeMo at https://github.com/NVIDIA/NeMo/blob/f1bf6c2731c6d4881b0063aa26564d9cd8300c4c/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L82.